### ISSUE TITLE
no functionality at all on latest version of VScode on windows 10

### ISSUE THREAD

**Issue by tensor-programming** [Dart-Code/Dart-Code#614]
As the title says: I am literally getting no functionality out of Dart-Code after the latest update from VSCode.  Things were working perfectly before the update but now no linting and only syntax highlighting is working.  I've tried resetting my settings and reinstalling the plugin to no avail.  I am running sdk 2.0.0-dev.20.0 on windows 10 if that helps.

**Comment by DanTup** [comment]
Yikes! I did install on Windows just after publishing to give it a quick test and it looked good, so hopefully it's something simple.

Can you click Help -> Show Developer Tools and check if there are any errors in the console?

Also if you [enable `dart. analyzerLogFile `](https://github.com/Dart-Code/Dart-Code/blob/master/CONTRIBUTING.md#enable-logging) and restart then that might give some hints.

Some other questions:

- Are you opening a Dart or Flutter project?
- Do you see a Dart SDK version in the status bar?
- You're on both VS Code 1.20.1 stable and Dart Code 2.9.0, right?

I might not be around much of the day, but feel free to ping in [Gitter chat](https://gitter.im/dart-code/Dart-Code) when you're looking at this and if I'm around we may be able to troubleshoot quicker there.

**Comment by tensor-programming** [comment]
So I am seeing errors in the show developer tools section.  

`Activating extension `Dart-Code.dart-code` failed: Cannot find module .vscode\extensions\Dart-Code.dart-code-2.9.0\out\src\extension'.` 

here is the stacktrace: 
```
at Function.Module._resolveFilename (module.js:470:15)
	at Function.Module._load (module.js:418:25)
	at Function.o._load (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:664:413)
	at Module.require (module.js:498:17)
	at require (internal/module.js:20:19)
	at Function.a [as __$__nodeRequire] (c:\Program Files\Microsoft VS Code\resources\app\out\vs\loader.js:31:863)
	at c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:687:367
	at t._doActivateExtension (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:687:479)
	at t._activateExtension (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:686:357)
	at Object.actualActivateExtension (c:\Program Files\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:684:415)
```

I am not seeing the dark SDK version in the status bar at all (thats how I knew it wasn't working properly off the bat.  I was opening a mixture of Flutter and Dart applications using code and it still doesn't seem to work.  Even today, I just updated flutter to the Main branch so that I could use the dart 2 features but I still can't seem to get code to work properly.  

I only have the default options setup for this extension; my SDK and Flutter SDK are on the path and I have the variables setup properly (As mentioned before, everything worked fine prior to the update).  I also am not seeing any errors when I open my project.  There is no linting at all anymore or auto-completion.  Used to be that when I saved a pubspec.yaml file it would automatically run pub get, now it does nothing.   

VS code version 1.20.1 and dart code is 2.9.0.  (Ive tired manually removing the extension and reinstalling to no avail.) 

Any help would be appreciated.

**Comment by DanTup** [comment]
I'm a bit confused by this! The first error you posted suggests it's failing to load the extension (which certainly would cause the behaviour you're seeing).

However, the second error shows the extension code running - I think the error there is because your `analyzerLogFile` might not be set to a string? It seems to be trying to call `.startsWith` in the value there (to resolve paths starting with `~`) but it doesn't exist.

Can you check your value of `dart. analyzerLogFile` and try again? I'm curious if you fix the second error does the first one still occur?

**Comment by DanTup** [comment]
Ok, I can repro the second error by setting `analyzerLogPath` to a non-string value (eg. `true`). Code isn't giving any help on this property because it's "hidden", so I'm going to try and do #413 for the next version.

Can you try setting the value to a string path (a log filename) and then see whether you get any errors? I think all of your last comment might be as a result of the log file path being wrong rather than the original error.

**Comment by tensor-programming** [comment]
Alright, so I checked and I did not have the `analyzerLogPath` variable set to anything.  However, on the default side of the panel it said it was set to a value of true... (this was the most bizarre issue I've ever personally encountered).  I have no idea how that was even possible but yes, correcting that did in fact correct the issue and now it seems the extension is running as intended.  

Given what you told me about the first error; the progression of events for this particular bug seems to be as follows.  I had no functionality from this extension after VScode updated to 1.20.1 as a result of something that remains to be unknown (perhaps a small error in the download etc.)  After reinstalling the extension and VS Code, this error was resolved but the new error arose as a result of the default value being changed? Honestly, I can't think of a way this could happen (because It would suggest  that other people would have the same issue).  

Anyhow, thank you for your help.   I love this extension, so keep up the good work.
